packages/jira-adapter/src/filters/request_type.ts

```diff
@@ -0,0 +1,180 @@
+* limitations under the License.
+*/
+
+import { AdditionChange, Change, InstanceElement, ModificationChange, ReferenceExpression, getChangeData, isAdditionChange, isAdditionOrModificationChange, isEqualValues, isInstanceChange, isModificationChange } from '@salto-io/adapter-api'
+import _ from 'lodash'
+import { values as lowerDashValues } from '@salto-io/lowerdash'
+import { getParent, isResolvedReferenceExpression } from '@salto-io/adapter-utils'
+import { FilterCreator } from '../filter'
+import { ISSUE_VIEW_TYPE, REQUEST_FORM_TYPE, REQUEST_TYPE_NAME } from '../constants'
+import { deployChanges, defaultDeployChange } from '../deployment/standard_deployment'
+import JiraClient from '../client/client'
+import { layoutConfigItem } from './layouts/layout_types'
+import { LayoutTypeName, getLayoutResponse, isIssueLayoutResponse, layoutTypeNameToDetails } from './layouts/layout_service_operations'
+
+const { isDefined } = lowerDashValues
+
+type statusType = {
+  id: ReferenceExpression
+  original: string
+  custom: boolean
+}
+
+const fieldsToIgnore = [
+  'issueView',
+  'requestForm',
+  'workflowStatuses',
+  'avatarId',
+  'groupIds',
+]
+
+
+const deployWorkflowStatuses = async (
+  change: ModificationChange<InstanceElement> | AdditionChange<InstanceElement>,
+  client: JiraClient): Promise<void> => {
+  if ((isAdditionChange(change))
+    || (isModificationChange(change)
+    && !isEqualValues(change.data.before.value.workflowStatuses, change.data.after.value.workflowStatuses))) {
+    const instance = getChangeData(change)
+    const parent = getParent(instance)
+    const workflowStatuses = instance.value.workflowStatuses.map((status: statusType) => ({
+      id: status.id?.value.value.id,
+      original: status.id?.value.value.name,
+      custom: status.custom,
+    }))
+    await client.post({
+      url: `/rest/servicedesk/1/servicedesk-data/${parent.value.key}/request-type/${instance.value.id}/workflow`,
+      data: workflowStatuses,
+    })
+  }
+}
+
+const isRequesTypeDetailsChange = (
+  change: ModificationChange<InstanceElement>,
+  fieldName: string,
+): boolean => fieldName === 'requestForm'
+  && (change.data.before.value.name !== change.data.after.value.name
+    || change.data.before.value.description !== change.data.after.value.description
+    || change.data.before.value.helpText !== change.data.after.value.helpText)
+
+
+const deployRequestTypeLayout = async (
+  change: Change<InstanceElement>,
+  client: JiraClient,
+  typeName: LayoutTypeName,
+): Promise<void> => {
+  const requestType = getChangeData(change)
+  const { fieldName } = layoutTypeNameToDetails[typeName]
+  if (fieldName === undefined) {
+    return undefined
+  }
+  if ((isAdditionChange(change))
+  || (isModificationChange(change)
+  && (!isEqualValues(change.data.before.value[fieldName], change.data.after.value[fieldName])
+  || isRequesTypeDetailsChange(change, fieldName)))) {
```

Comment:
readability nit - especially in complex conditions with and+or at different levels, indentation can really help :) 
```suggestion
  if (
    isAdditionChange(change)
    || (isModificationChange(change)
      && (
        !isEqualValues(change.data.before.value[fieldName], change.data.after.value[fieldName])
        || isRequesTypeDetailsChange(change, fieldName)
      ))
  ) {
```
(or some other alternative, but clarifying where a compound condition begins and where it ends)